### ISSUE TITLE
Banner hide/fade code confusion

### ISSUE THREAD

**Issue by bobular** [VEuPathDB/CoreUI#147]
Reviewing a lineplot PR I noticed that `Banner`'s `banner.setShowBanner` is an optional prop.

It's only used when `autoHideDuration` is truthy.  I guess this is OK, but the docs should say that `setShowBanner` is required to make auto-hide work.

I went down a rabbit hole of finding out what `setTimeout(func, undefined)` does and I think the undefined is coerced into zero, which means it (actually a no-op) is executed straight away.

I think it might be better just to start the useEffect (see below) with `if (autoHideDuration == null) return;`

Also the 1 second fade-out isn't well documented. If `autoHideDuration` is less than 1000 (the prop docs don't say milliseconds btw) then what happens with a negative timeout duration? It's probably no big deal, but we are relying on poorly documented edge-case behaviours of built-in JS functions. It would be nicer to handle these ourselves.
```
  // Banner timeout with fadeout
  useEffect(() => {
    const autoFadeoutDuration = autoHideDuration ? autoHideDuration - 1000 : undefined;
    const fadeoutTimeout = setTimeout(() => {
      if (autoHideDuration && setFadeoutEffect) setFadeoutEffect(true);
    }, autoFadeoutDuration);

    const timeout = setTimeout(() => {
      if (autoHideDuration && setShowBanner) setShowBanner(false);
    }, autoHideDuration);
    return () => {
      clearTimeout(timeout);
      clearTimeout(fadeoutTimeout);
    };
  }, [showBanner, autoHideDuration, fadeoutEffect]);
```

**Comment by dmfalke** [comment]
`setTimeout(func, 0)` will schedule the invocation of `func` for the next event loop. It will not execute right away

**Comment by dmfalke** [comment]
Do we have a know use case for automatic hiding banners? I thought we were going to use notifications (I don't remember what we call them internally)

**Comment by dmfalke** [comment]
SnackBar!

**Comment by bobular** [comment]
Yes, snackbar, and yes, I meant "effectively straight away", sorry!
We can discuss on Wednesday removing the auto-hide facility?
